src/main/java/school/lemon/changerequest/java/introduction/hw2/Task9.java

```diff
@@ -10,19 +10,18 @@
         }
 
 
-        String shema = "([a-z]+(?=\\:\\/\\/))";  // for all leters before ://
-        String host = "(([a-z0-9.-_])+ (?=\\:))";  // for all leters , numbers and . -_ before :  (but if url dont have :port ?)
-        String port = "(?=\\:[0-9]+)";  // for all numbers after :
-        Pattern pshema = Pattern.compile(shema);
-        Pattern phost = Pattern.compile(host);
-        Pattern pport = Pattern.compile(port);
-        Matcher mshema = pshema.matcher(URL);
-        Matcher mhost = phost.matcher(URL);
-        Matcher mport = pport.matcher(URL);
 
+String url =" ( (?<=\\:\\/\\/) [\\w] +.) ? ((?<= \\/\\/) [\\w\\d.-_] +.) ((?<=\\: ) [\\d] ) ";
+        Pattern purl = Pattern.compile(url);
+        Matcher murl = purl.matcher(URL);
+        StringBuilder surl = new StringBuilder();
+        if (murl.find()){
+           surl.append(murl.group()).append("\n");
+        return new String[]{surl.toString()};
```

Comment:
Try to avoid such naming.
Characters are free of charge now and you can easily create names with any length.
Also, follow lowerCamelCase for variables names.
Following names would be better:
`url -> regexp`
`purl -> urlPattern`
`murl -> urlMatcher`
`surl -> result`.
Also, you need to provide `String[3]` array and with current implementation you'll supply `String[1]` array, as `surl.toString()` will create one `String` object.